src/components/withEnvironment.js

```diff
@@ -0,0 +1,46 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import Environment from '../libs/Environment';
+import CONST from '../CONST';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+const environmentPropTypes = {
+    /** The string value representing the current environment */
+    environment: PropTypes.string.isRequired,
+};
+
+export default function (WrappedComponent) {
```

Comment:
NABish, but I think we should generally make all our HOCs forward refs to the wrapped component. It's a common pitfall to pass a ref to a component that's wrapped in a HOC, only to find that the HOC "swallows" the ref. There are examples of how to forward refs in a HOC in [`withLocalize`](https://github.com/Expensify/Expensify.cash/blob/main/src/components/withLocalize.js) and [`withWindowDimensions`](https://github.com/Expensify/Expensify.cash/blob/1c1471290a419d923fd1980d24bc0b43be888785/src/components/withWindowDimensions.js#L92-L95).

Reply:
Great point, I agree we should get in the habit of doing this with all HOCs and I'd like to see it done in this PR.

Reply:
Sounds good, I'll check the other HOCs and add it if its missing in any others as well